internal/mining/mining_harness_test.go

```diff
@@ -79,6 +79,54 @@ type fakeChain struct {
 	utxos                              *blockchain.UtxoViewpoint
 }
 
+// Returns a new fakeChain to be used for testing.
+// When using this function to create a fakeChain to pass to newMiningHarness(...), it is
+// recommended that the same chainParams used to call this function is used to call
+// newMiningHarness.
+// If the value of heightOfBestState is less than chainParams.CoinbaseMaturity, it will be
+// set to chainParams.CoinbaseMaturity
+func NewFakeChain(chainParams *chaincfg.Params, heightOfBestState int64) *fakeChain {
+	if heightOfBestState < int64(chainParams.CoinbaseMaturity) {
+		heightOfBestState = int64(chainParams.CoinbaseMaturity)
+	}
```

Comment:
I'll comment on the overall design too, so this may not matter, but I don't see any reason why you'd not allow setting a best height less than coinbase maturity.

Reply:
> Do you happen to know why it's doing `int64(chainParams.CoinbaseMaturity) + curHeight` and not just setting it to `curHeight`? Maybe I can just remove this dependency between the two and see if some test breaks. I just thought there was a reason behind it.

The only reason it's setting it to `int64(chainParams.CoinbaseMaturity) + curHeight` because it's creating an output from a coinbase and a coinbase output is not spendable until `chainParams.CoinbaseMaturity`, and the intention is to return spendable outputs for the tests to start with.